### DISCUSSION TITLE
Installation failed on FlowFuse Cloud

### DISCUSSION THREAD

**Discussion by gabrielboboc2** [GogoVega/node-red-contrib-firebase-realtime-database#62] [Q&A] [answered]
Is it possible to make it compatible with Flowfuse, which is an application similar to node red? I've tried to install the library, but apparently there is a problem because the installation failed, and so I thought  maby it do not have some permitions.
Thank you very much.

**Comment by GogoVega** [comment]
Hi,
FlowFuse allows you to control Node-RED instance(s), so it's compatible 😉
I just tested locally and had no problems.
Please could you send me the logs (Node-RED logs section)

**Reply by gabrielboboc2** [reply]
-----------------------------------------------------------
2023-12-13T21:17:58.732Z Install : @gogovega/node-red-contrib-firebase-realtime-database 0.5.3

2023-12-13T21:17:58.579Z npm install --no-audit --no-update-notifier --no-fund --save --save-prefix=~ --production --engine-strict @gogovega/node-red-contrib-firebase-realtime-database@0.5.3
2023-12-13T21:18:03.984Z [err] npm
2023-12-13T21:18:03.985Z [err]  WARN config
2023-12-13T21:18:03.985Z [err]  production Use `--omit=dev` instead.
2023-12-13T21:18:35.981Z rc=null

**Reply by gabrielboboc2** [reply]
this is the log

**Reply by GogoVega** [reply]
Unfortunately it doesn't help me much, what device do you have, is it a Docker?

This may be related to a RAM problem, you can try install it manually by running the following command:

```bash
cd /opt/flowforge/var/projects/your-instance-id
npm i @gogovega/node-red-contrib-firebase-realtime-database@latest --omit=dev
```

Then restart your Node-RED instance

**Reply by gabrielboboc2** [reply]
FlowFuse ce n'est pas l'application Node RED qui est installée à l'aide de la docker. C'est une application semblable à Node RED, la seule différence c'est que c'est une application qui est complètement sur le serveur.
Je trouve bien ta librairie, mais lors de l'installation, il me montre que l'installation a échué.
Je sais pas si tu peux m'aider dans ce sens-là.
Mais je te remercie qund meme.

**Reply by GogoVega** [reply]
Comme je disais FlowFuse est un serveur te permettant de gérer des instances Node-RED qu'elles soient créées par FlowFuse ou utilisées comme appareil.

Si t'as installé FlowFuse sans Docker, ton dossier se trouve soit dans `/opt/flowforge` soit dans `c:\flowforge`
Tu peux donc lancer une installation manuelle avec la commande précédente

Si tu utilises Docker, tu dois lancer la commande depuis le Docker mais je ne connais pas le chemin exacte, c'est normalement quelque chose comme var/projects. Là aussi utilises la commande précédente en modifiant le chemin.

Tkt, on va trouver une solution 😜

**Reply by gabrielboboc2** [reply]
En fait je n'ai rien installé, j'ai juste cree un cont sur web sur https://flowfuse.com/, et la je sais créé des instances Node red complètement sur web.

**Comment by GogoVega** [comment, marked as answer]
So, if you use the FlowFuse Cloud version (Starter) you risk having an error during installation.
This is due to the limited RAM size and the installation exceeding this limit is killed.

To resolve this issue and be able to install this module follow the following steps:
1. Suspend your instance
2. Go to your instance settings
3. Select "Palette" then go down to the "Installed Modules" section
4. Click on "Add module"
5. Write `@gogovega/node-red-contrib-firebase-realtime-database` and `~0.5.4` (0.5.4 is the current latest version)
6. Click on "Save settings"
7. Start your instance (your instance will start after installing dependencies)
